<?xml version="1.0"?>
<!--
  Launch calculation of individual magnetic vectors
-->
<launch>
  


<!-- run node -->
  <node name="magnetic_field_vector1" pkg="magnetic_field_localization" type="magnetic_field_vector_fast" output="screen">
    <param name="magnetometer_topic" type="str" value="/imu1/mag" />
    <param name="vector_topic" type="str" value="/magnetic_vector0" />
    <param name="magnetometer_frame" type="str" value="/magnetometer0" />
    <param name="magnetic_field_frequency" type="int" value="50" />
    <param name="number_of_points_for_analysis" type="int" value="20" />
    <param name="number_of_cycles_for_analysis" type="int" value="1" />
    <param name="magnetometer_sampling_time" type="double" value="0.01288" />
  </node>

  <node name="magnetic_field_vector2" pkg="magnetic_field_localization" type="magnetic_field_vector_fast" output="screen">
    <param name="magnetometer_topic" type="str" value="/imu2/mag" />
    <param name="vector_topic" type="str" value="/magnetic_vector1" />
    <param name="magnetometer_frame" type="str" value="/magnetometer1" />
    <param name="magnetic_field_frequency" type="int" value="50" />
    <param name="number_of_points_for_analysis" type="int" value="20" />
    <param name="number_of_cycles_for_analysis" type="int" value="1" />
    <param name="magnetometer_sampling_time" type="double" value="0.01282" />
  </node>
  <node name="magnetic_field_vector3" pkg="magnetic_field_localization" type="magnetic_field_vector_fast" output="screen">
    <param name="magnetometer_topic" type="str" value="/imu3/mag" />
    <param name="vector_topic" type="str" value="/magnetic_vector2" />
    <param name="magnetometer_frame" type="str" value="/magnetometer2" />
    <param name="magnetic_field_frequency" type="int" value="50" />
    <param name="number_of_points_for_analysis" type="int" value="20" />
    <param name="number_of_cycles_for_analysis" type="int" value="1" />
    <param name="magnetometer_sampling_time" type="double" value="0.0128" />
  </node>

  <node pkg="tf" type="static_transform_publisher" name="magnetometer_position" args="0 -0.3 0 0 0 0 /magnetometer0 /magnetometer1 100"/> 
  <node pkg="tf" type="static_transform_publisher" name="magnetometer_position1" args="0 -0.3 0 0 0 0 /magnetometer1 /magnetometer2 100"/> 
  <node pkg="tf" type="static_transform_publisher" name="magn" args="-0.167 -0.028 0 -0.8 0 0 /ee_link /magnetometer0 100"/> 

</launch>
